vm/page.c

```diff
@@ -120,28 +124,34 @@ void _page_free(page_t *p)
 		pages.allocsz -= SIZE_PAGE;
 	}
 
-	if (p->addr & ((1 << (idx + 1)) - 1))
+	if ((p->addr & ((1 << (idx + 1)) - 1)) != 0) {
 		lh = p - (1 << idx) / SIZE_PAGE;
-	else
+	}
+	else {
 		rh = p + (1 << idx) / SIZE_PAGE;
+	}
 
-	while (lh >= pages.pages && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && (lh->flags & PAGE_FREE) && (rh->flags & PAGE_FREE) && (lh->idx == rh->idx) && (lh->addr + (1 << lh->idx) == rh->addr) && (idx < SIZE_VM_SIZES)) {
+	while ((lh >= pages.pages) && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && (lh->flags & PAGE_FREE) && ((rh->flags & PAGE_FREE) != 0) && (lh->idx == rh->idx) && (lh->addr + (1 << lh->idx) == rh->addr) && (idx < SIZE_VM_SIZES)) {
```

Comment:
```suggestion
	while ((lh >= pages.pages) && (rh < pages.pages + (pages.allocsz + pages.freesz) / SIZE_PAGE) && ((lh->flags & PAGE_FREE) != 0) && ((rh->flags & PAGE_FREE) != 0) && (lh->idx == rh->idx) && ((lh->addr + (1 << lh->idx)) == rh->addr) && (idx < SIZE_VM_SIZES)) {
```

Reply:
I know I'm supper nitpicky, but I think you missed:
is:
`(lh->addr + (1 << lh->idx) == rh->addr)`
should be:
`((lh->addr + (1 << lh->idx)) == rh->addr)`